src/content/api/parser.md

```diff
@@ -37,279 +38,477 @@ as such:
+General purpose hook that is called for every parsed statement in a code fragment.
+
+- Callback Parameters: `statement`
+
+```js
+parser.hooks.statement.tap('MyPlugin', statement => { /* ... */ });
+```
 
-Parameters: `statement`
+Where the `statement.type` could be:
+
+- `'BlockStatement'`
+- `'VariableDeclaration'`
+- `'FunctionDeclaration'`
+- `'ReturnStatement'`
+- `'ClassDeclaration'`
+- `'ExpressionStatement'`
+- `'ImportDeclaration'`
+- `'ExportAllDeclaration'`
+- `'ExportDefaultDeclaration'`
+- `'ExportNamedDeclaration'`
+- `'IfStatement'`
+- `'SwitchStatement'`
+- `'ForInStatement'`
+- `'ForOfStatement'`
+- `'ForStatement'`
+- `'WhileStatement'`
+- `'DoWhileStatement'`
+- `'ThrowStatement'`
+- `'TryStatement'`
+- `'LabeledStatement'`
+- `'WithStatement'`
 
 
 ### statementIf
 
 `SyncBailHook`
 
-...
+Called when parsing an if statement. Same as the `statement` hook, but triggered only when `statement.type == "IfStatement"`
 
-Parameters: `statement`
+- Callback Parameters: `statement`
 
 
 ### label
 
 `SyncBailHook`
 
-...
+Called when parsing statements with a [label](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/label). Those statements have `statement.type === 'LabeledStatement'`.
 
-Parameters: `statement`
+- Hook Parameters: `labelName`
+
+- Callback Parameters: `statement`
 
 
 ### import
 
 `SyncBailHook`
 
-...
+Called for every import statement in a code fragment. The `source` parameter contains the name of the imported file.
+
+- Callback Parameters: `statement` `source`
 
-Parameters: `statement` `source`
+```js
+// index.js
+import _ from 'lodash';
+
+// MyPlugin.js
+parser.hooks.import.tap('MyPlugin', (statement, source) => {
+  /* source == 'lodash' */
+});
+```
 
 
 ### importSpecifier
 
 `SyncBailHook`
 
-...
+Called for every specifier of every `import` statement.
+
+- Callback Parameters: `statement` `source` `exportName` `identifierName`
 
-Parameters: `statement` `source` `exportName` `identifierName`
+```js
+// index.js
+import _, { has } from 'lodash';
+
+// MyPlugin.js
+parser.hooks.import.tap('MyPlugin', (statement, source, exportName, identifierName) => {
+  /* First call
+    source == 'lodash'
+    exportName == 'default'
+    identifierName == '_'
+  */
+  /* Second call
+    source == 'lodash'
+    exportName == 'has'
+    identifierName == 'has'
+  */
+});
+```
 
 
 ### export
 
 `SyncBailHook`
 
-...
+Called for every `export` statement in a code fragment.
 
-Parameters: `statement`
+- Callback Parameters: `statement`
 
 
 ### exportImport
 
 `SyncBailHook`
 
-...
+Called for every `export`-import statement eg: `export * from 'otherModule';`.
 
-Parameters: `statement` `source`
+- Callback Parameters: `statement` `source`
 
 
 ### exportDeclaration
 
 `SyncBailHook`
 
-...
+Called for every `export` statement exporting a declaration.
+
+- Callback Parameters: `statement` `declaration`
+
 
-Parameters: `statement` `declaration`
+```js
+// Those will trigger the hook
```

Comment:
Suggest to take out of js snippet and change to `These exports will trigger the hook: ...`